### ISSUE TITLE
pivot-root support in portablectl / portabled

### ISSUE THREAD

**Issue by davide125** [systemd/systemd#13094]
**Is your feature request related to a problem? Please describe.**
I have a thing building btrfs filesystem images I'd like to use as portable services that have subvolumes in them

**Describe the solution you'd like**
- portablectl attach/inspect/etc would optionally take a `--subvolume` option to choose which btrfs subvolume should be used
- (maybe) if `--subvolume` is not used, *and* there is only a single subvolume, portablectl attach/inspect/etc would automatically use the subvolume

**Describe alternatives you've considered**
I could modify the build system for the btrfs filesystem images to call `btrfs subvolume set-default` (and I probably will), but I think this would be a useful feature in portablectl itself regardless.

**Comment by poettering** [comment]
Hmm, but we optionally mount multiple file systems from a single image (i.e. there are well-known partitions for /home, /srv), so we'd have to make that path configurable for each?

I figure this is not specific to btrfs though, I mean, btrfs subvols are located in the top-level root dir, so we could generalize this as a way to mount a subdir of the actual root, if necessary...

I figure this could work like the existing --pivot-root= for systemd-nspawn, which sounds like it would do what you want?

**Comment by davide125** [comment]
Yes, something like `--pivot-root=` would be perfect here, and nicely avoid any making this needlessly btrfs-specific.
